### PR TITLE
PH-1140: :recycle: Refactor app_developer_mode feature flag

### PR DESCRIPTION
**Pull request by jmleroux**
**Description (for Contributor and Core Developer)**

This flag should be enabled from the PRODUCT_REFERENCE_CODE

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/FeatureFlag/AppDeveloperModeFeature.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+use Akeneo\Platform\Bundle\FeatureFlagBundle\FeatureFlag;
+use Akeneo\Platform\Bundle\FeatureFlagBundle\FeatureFlags;
+
+/**
+ * @copyright 2023 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ * @author    JM Leroux <jean-marie.leroux@akeneo.com>
+ */
+final class AppDeveloperModeFeature implements FeatureFlag
```
Comment (ghoussard) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/FeatureFlag/AppDeveloperModeFeature.php:15]: I'd name it like that to be coherent with others feature flag services. Here the service name is too related with a specific feature
```suggestion
final class OnlySaasDevFeatureFlag implements FeatureFlag
```

Comment (jmleroux) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/FeatureFlag/AppDeveloperModeFeature.php:15]: I took the first example I found :| ...

https://github.com/akeneo/pim-community-dev/blob/fa1e683fdbda3595e5ec4c90bd075bee15f79585/src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/FeatureFlag/AllCriteriaFeature.php#L21

Comment (ghoussard) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/FeatureFlag/AppDeveloperModeFeature.php:15]: Indeed! So, why not put the class in FeatureFlag directory and just keep the service declaration in connectivity?

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/FeatureFlag/AppDeveloperModeFeature.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+{
+    private const PRODUCT_REFERENCE_CODES = [
+        'serenity_dev',
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/FeatureFlag/AppDeveloperModeFeature.php:18]: Does it mean that the deployed non-production instances will have these values?

Comment (jmleroux) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/FeatureFlag/AppDeveloperModeFeature.php:18]: yes, the `PRODUCT_REFERENCE_CODE` is an available env var

**Comment by jmleroux** [comment]
Closing because will be totally removed

**Pull request closed without merging**